### PR TITLE
Modifications for receiver with optical inputs

### PR DESCRIPTION
**Pull request by phHartl**
Tested on a fresh installation with an R-N500. The corresponding pull request is also available for the internal python library. I have not executed the tests, as I modified the code directly on my HA installation with a restricted Python environment. However, I tried to mirror the changes of previous PRs (#294) that included new sources.

## Summary by CodeRabbit

- **New Features**
	- Introduced new input mappings for OPTICAL1 and OPTICAL2.
	- Updated configuration flow to support the new minor version.

- **Bug Fixes**
	- Improved migration logic to handle transitions between versions more effectively.

- **Documentation**
	- Updated requirements to reflect the latest dependency version.

- **Tests**
	- Added new tests for migration and configuration options, ensuring robustness of new features.
	- Adjusted existing tests to align with the updated migration logic and new inputs.

### CODE REVIEW

File: custom_components/yamaha_ynca/input_helpers.py [hunk 1]
```diff
@@ -54,6 +54,8 @@ class Mapping:
     Mapping(ynca.Input.HDMI5, []),
     Mapping(ynca.Input.HDMI6, []),
     Mapping(ynca.Input.HDMI7, []),
+    Mapping(ynca.Input.OPTICAL1, []),
+    Mapping(ynca.Input.OPTICAL2, []),
```
Comment (mvdwetering) [inline custom_components/yamaha_ynca/input_helpers.py:58]: Could you move these to below MULTICH so the list stays alphabetical?

Comment (phHartl) [reply custom_components/yamaha_ynca/input_helpers.py:59]: Move it below MULTICH and switched USB and VAUX, so the list is truly alphabetical.

**Comment by mvdwetering** [comment]
Thanks for the pullrequest.

Would you mind adding the diagnostics file to this PR?
I am interested in the `initialization` section because the R-N500 is the first device reported working from that range. The others devices all seem to be AV receivers while the R-N500 seems to be audio only, so it would be interesting to get some more insights in what capabilities it supports/reports.

**Comment by mvdwetering** [comment]
btw, sorry for the amount of work it is to add an additional input. It is on my list to make this simpler, but did not have time yet 😞

**Comment by phHartl** [comment]
> Thanks for the pullrequest.
> 
> Would you mind adding the diagnostics file to this PR? I am interested in the `initialization` section because the R-N500 is the first device reported working from that range. The others devices all seem to be AV receivers while the R-N500 seems to be audio only, so it would be interesting to get some more insights in what capabilities it supports/reports.

I checked out the complete repository on my laptop now and fixed the tests using the pytests module. What exactly do you mean by 'initialization' section, and how can I get this information? I'm still new to Home Assistant in general.

**Comment by phHartl** [comment]
> I was referring to the "Download diagnostics" option that some/most integrations offer nowadays. It will download a file which typically contains more detailed data useful for troubleshooting.
> 
> For this integration I store all the commands sent during initialization and the most recent commands.
> 
> It helps me to see what is going on and I can load those commands in my little simulator so I can often reproduce the issues people have without access to the actual device. I am interested since the R-N500 is the first device I know that works with this protocal that is not an AV receiver so it is nice to have the commands it supports as a reference.
> 
> image

Thanks for the explanation, the corresponding output is:

<details>
<summary>Click to expand</summary>

```json
"initialization": [
        "Send: @SYS:MODELNAME=?",
        "Received: @SYS:MODELNAME=R-N500",
        "Send: @SYS:MODELNAME=?",
        "Received: @SYS:MODELNAME=R-N500",
        "Send: @SYS:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:AVAIL=?",
        "Received: @MAIN:AVAIL=Ready",
        "Send: @ZONE2:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @ZONE3:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @ZONE4:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @AIRPLAY:AVAIL=?",
        "Received: @AIRPLAY:AVAIL=Not Ready",
        "Send: @BT:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @DAB:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @IPOD:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @IPODUSB:AVAIL=?",
        "Received: @IPODUSB:AVAIL=Not Ready",
        "Send: @NAPSTER:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @NETRADIO:AVAIL=?",
        "Received: @NETRADIO:AVAIL=Not Ready",
        "Send: @PANDORA:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @PC:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @RHAP:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @SIRIUS:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @SIRIUSIR:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @SIRIUSXM:AVAIL=?",
        "Received: @RESTRICTED",
        "Send: @SERVER:AVAIL=?",
        "Received: @SERVER:AVAIL=Not Ready",
        "Send: @SPOTIFY:AVAIL=?",
        "Received: @SPOTIFY:AVAIL=Not Ready",
        "Send: @TUN:AVAIL=?",
        "Received: @TUN:AVAIL=Not Ready",
        "Send: @UAW:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @USB:AVAIL=?",
        "Received: @USB:AVAIL=Not Ready",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @SYS:AVAIL=?",
        "Received: @UNDEFINED",
        "Send: @SYS:HDMIOUT1=?",
        "Received: @RESTRICTED",
        "Send: @SYS:HDMIOUT2=?",
        "Received: @RESTRICTED",
        "Send: @SYS:HDMIOUT3=?",
        "Received: @UNDEFINED",
        "Send: @SYS:INPNAME=?",
        "Received: @RESTRICTED",
        "Send: @SYS:MODELNAME=?",
        "Received: @SYS:MODELNAME=R-N500",
        "Send: @SYS:PARTY=?",
        "Received: @RESTRICTED",
        "Send: @SYS:PARTYMUTE=?",
        "Received: @RESTRICTED",
        "Send: @SYS:PWR=?",
        "Received: @SYS:PWR=Standby",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @AIRPLAY:METAINFO=?",
        "Received: @AIRPLAY:ARTIST=",
        "Received: @AIRPLAY:ALBUM=",
        "Received: @AIRPLAY:SONG=",
        "Send: @AIRPLAY:AVAIL=?",
        "Received: @AIRPLAY:AVAIL=Not Ready",
        "Send: @AIRPLAY:PLAYBACKINFO=?",
        "Received: @AIRPLAY:PLAYBACKINFO=Stop",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @IPODUSB:METAINFO=?",
        "Received: @IPODUSB:ARTIST=",
        "Received: @IPODUSB:ALBUM=",
        "Received: @IPODUSB:SONG=",
        "Send: @IPODUSB:AVAIL=?",
        "Received: @IPODUSB:AVAIL=Not Ready",
        "Send: @IPODUSB:PLAYBACKINFO=?",
        "Received: @IPODUSB:PLAYBACKINFO=Stop",
        "Send: @IPODUSB:REPEAT=?",
        "Received: @IPODUSB:REPEAT=Off",
        "Send: @IPODUSB:SHUFFLE=?",
        "Received: @IPODUSB:SHUFFLE=Off",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @MAIN:ADAPTIVEDRC=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:AVAIL=?",
        "Received: @MAIN:AVAIL=Ready",
        "Send: @MAIN:ENHANCER=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:HDMIOUT=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:HPBASS=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:HPTREBLE=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:INITVOLLVL=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:INITVOLMODE=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:BASIC=?",
        "Received: @MAIN:PWR=Standby",
        "Received: @MAIN:SLEEP=Off",
        "Received: @MAIN:VOL=-49.5",
        "Received: @MAIN:MUTE=Off",
        "Received: @MAIN:INP=Spotify",
        "Send: @MAIN:LIPSYNCHDMIOUT1OFFSET=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:LIPSYNCHDMIOUT2OFFSET=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:MAXVOL=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:SCENENAME=?",
        "Received: @MAIN:SCENE1NAME=BD/DVD",
        "Received: @MAIN:SCENE2NAME=TV",
        "Received: @MAIN:SCENE3NAME=NET",
        "Received: @MAIN:SCENE4NAME=RADIO",
        "Send: @MAIN:SLEEP=?",
        "Received: @MAIN:SLEEP=Off",
        "Send: @MAIN:SPBASS=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:SPEAKERA=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:SPEAKERB=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:SPTREBLE=?",
        "Received: @UNDEFINED",
        "Send: @MAIN:3DCINEMA=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:2CHDECODER=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:ZONEBNAME=?",
        "Received: @RESTRICTED",
        "Send: @MAIN:ZONENAME=?",
        "Received: @MAIN:ZONENAME=Main",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @NETRADIO:METAINFO=?",
        "Received: @NETRADIO:STATION=",
        "Received: @NETRADIO:ALBUM=",
        "Received: @NETRADIO:SONG=",
        "Send: @NETRADIO:AVAIL=?",
        "Received: @NETRADIO:AVAIL=Not Ready",
        "Send: @NETRADIO:PLAYBACKINFO=?",
        "Received: @NETRADIO:PLAYBACKINFO=Stop",
        "Send: @NETRADIO:PRESET=?",
        "Received: @RESTRICTED",
        "Send: @NETRADIO:STATION=?",
        "Received: @NETRADIO:STATION=",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @SERVER:METAINFO=?",
        "Received: @SERVER:ARTIST=",
        "Received: @SERVER:ALBUM=",
        "Received: @SERVER:SONG=",
        "Send: @SERVER:AVAIL=?",
        "Received: @SERVER:AVAIL=Not Ready",
        "Send: @SERVER:PLAYBACKINFO=?",
        "Received: @SERVER:PLAYBACKINFO=Stop",
        "Send: @SERVER:REPEAT=?",
        "Received: @SERVER:REPEAT=Off",
        "Send: @SERVER:SHUFFLE=?",
        "Received: @SERVER:SHUFFLE=Off",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @SPOTIFY:METAINFO=?",
        "Received: @SPOTIFY:ARTIST=",
        "Received: @SPOTIFY:ALBUM=",
        "Received: @SPOTIFY:TRACK=",
        "Send: @SPOTIFY:AVAIL=?",
        "Received: @SPOTIFY:AVAIL=Not Ready",
        "Send: @SPOTIFY:PLAYBACKINFO=?",
        "Received: @SPOTIFY:PLAYBACKINFO=Stop",
        "Send: @SPOTIFY:REPEAT=?",
        "Received: @RESTRICTED",
        "Send: @SPOTIFY:SHUFFLE=?",
        "Received: @RESTRICTED",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @TUN:AMFREQ=?",
        "Received: @TUN:AMFREQ=801",
        "Send: @TUN:AVAIL=?",
        "Received: @TUN:AVAIL=Not Ready",
        "Send: @TUN:BAND=?",
        "Received: @TUN:BAND=FM",
        "Send: @TUN:FMFREQ=?",
        "Received: @TUN:FMFREQ=99.20",
        "Send: @TUN:PRESET=?",
        "Received: @TUN:PRESET=No Preset",
        "Send: @TUN:RDSINFO=?",
        "Received: @TUN:RDSPRGTYPE=",
        "Received: @TUN:RDSPRGSERVICE=",
        "Received: @TUN:RDSTXTA=",
        "Received: @TUN:RDSTXTB=",
        "Received: @TUN:RDSCLOCK=",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06",
        "Send: @USB:METAINFO=?",
        "Received: @USB:ARTIST=",
        "Received: @USB:ALBUM=",
        "Received: @USB:SONG=",
        "Send: @USB:AVAIL=?",
        "Received: @USB:AVAIL=Not Ready",
        "Send: @USB:PLAYBACKINFO=?",
        "Received: @USB:PLAYBACKINFO=Stop",
        "Send: @USB:PRESET=?",
        "Received: @RESTRICTED",
        "Send: @USB:REPEAT=?",
        "Received: @USB:REPEAT=Off",
        "Send: @USB:SHUFFLE=?",
        "Received: @USB:SHUFFLE=Off",
        "Send: @SYS:VERSION=?",
        "Received: @SYS:VERSION=1.17/2.06"
      ],
</details> ```

This should be what you are looking for, right? Btw, don't worry about the work one needs to do to add another input. It is not that much :)

**Pull request merged**